src/jshint.js

```diff
@@ -5011,11 +4804,7 @@ var JSHINT = (function() {
     }
   }
 
-  var warnUnused = function(name, tkn, type, unused_opt) {
-    var line = tkn.line;
-    var chr  = tkn.from;
-    var raw_name = tkn.raw_text || name;
-
+  var getUnusedOption = function(unused_opt) {
```

Comment:
I like this refactoring, but it introduces yet another function that closes around the global `state` variable. A `state.getUnused` method in `state.js` would make the function's dependency on the `state` object much more obvious, and it would be a more discoverable location for the logic. What do you think?